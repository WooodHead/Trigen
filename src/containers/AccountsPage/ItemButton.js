import styled from 'styled-components';

import Button from 'components/Button';


const getBGImage = props => {
  switch(props.mode) {
    case 'save':
      return `${props.theme.primary} url("src/img/icons/Icon_Button_Save.png") center/14px 14px`;
    case 'cancel':
      return `${props.theme.secondary} url("src/img/icons/Icon_Button_Abort.png") center/15px 15px`;
    case 'edit':
      return `${props.theme.primary} url("src/img/icons/Icon_Button_Edit.png") center/15px 15px`;
    case 'delete':
      return `${props.theme.secondary} url("src/img/icons/Icon_Button_Delete.png") center/11px 15px`;
    default:
      return `${props.theme.secondary} url("src/im/icons/Icon_Button_Abort.png") center/15px 15px`;
  }
};

export default styled(Button)`
  margin: 0 5px 0 0;
  width: 23px;
  height: 23px;

  border-radius: 50%;
  background: ${props => getBGImage(props)} no-repeat;
`;
