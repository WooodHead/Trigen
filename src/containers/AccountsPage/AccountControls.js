import React, { PropTypes } from 'react';
import styled from 'styled-components';

import ItemButton from './ItemButton';


const Wrapper = styled.div``;

const AccountControls = props => {
  return (
    <Wrapper>
      {
        props.edit
          ? ([
              <ItemButton key="save" onClick={props.save} mode="save" />,
              <ItemButton key="cancel" warning onClick={props.cancel} mode="cancel" />
            ]) : ([
              <ItemButton key="edit" onClick={props.editItem} mode="edit" />,
              <ItemButton key="delete" warning onClick={props.delete} mode="delete" />
            ])
      }
    </Wrapper>
  );
}

export default AccountControls;
