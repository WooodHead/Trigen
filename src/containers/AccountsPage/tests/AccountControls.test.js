import  React from 'react';
import { shallow } from 'enzyme';

import AccountControls from '../AccountControls';
import ItemButton from '../ItemButton';


function setUp(newProps) {
  const props = {
    save: jest.fn(),
    cancel: jest.fn(),
    editItem: jest.fn(),
    delete: jest.fn(),
    edit: false,
    ...newProps
  }

  const enzymeWrapper = shallow(<AccountControls {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('<AccountControls', () => {
  it('should render edit <ItemButton />', () => {
    const { enzymeWrapper } = setUp();

    expect(enzymeWrapper.find(ItemButton).findWhere(b => b.props().mode === 'edit').length).toMatchSnapshot();
  });

  it('should render Delete <ItemButton />', () => {
    const { enzymeWrapper } = setUp();

    expect(enzymeWrapper.find(ItemButton).findWhere(b => b.props().mode === 'delete').length).toMatchSnapshot();
  });
});
