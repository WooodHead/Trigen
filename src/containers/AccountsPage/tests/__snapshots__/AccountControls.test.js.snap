exports[`<AccountControls should render Delete <ItemButton /> 1`] = `1`;

exports[`<AccountControls should render edit <ItemButton /> 1`] = `1`;
